enhancements/installer/single-node-installation.md

```diff
@@ -0,0 +1,192 @@
+---
+title: Single node installation
+authors:
+  - "@eranco74"
+reviewers:
+  - "@romfreiman"
+  - "@hardys"
+  - "@crawford"
+  - "@markmc"
+approvers:
+  - "@hardys"
+  - "@crawford"
+  - "@markmc"
+creation-date: 2020-08-18
+last-updated: 2020-08-30
+status: provisional
+see-also:
+  - "https://github.com/openshift/enhancements/pull/302"
+  - "https://github.com/openshift/installer/pull/3978"
+---
+
+
+# Single node installation
+
+Add a new `create single-node-config` command to `openshift-installer` which
+allows a user to create an `aio.ign` Ignition configuration which
+launches a minimal all-in-one control plane using static pods.
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+The new installer `create single-node-config` command generates an `aio.ign`
+file with a minimal all-in-one control plane using static pods. The
+installer command will use the same `install-config.yaml` input as the
+`create cluster` command.
+
+This minimal control plane is similar to the bootstrap control plane
+used in regular installations, and consists only of etcd,
+kube-apiserver, kube-controller-manager, and kube-scheduler. The key
+differences from the usual `bootstrap.ign` configuration are:
+
+- `bootkube.sh` will finish when it sets up the control plane without
+  applying all the OpenShift manifests.
+- `kubelet` will be configured to be a part of the cluster.
+
+When a user boots a CoreOS image with the `aio.ign` configuration,
+they will observe a similar flow as a bootstrap node in a regular
+installation, but they will end up with a minimal control plane. The
+intent is that the user will be able to expand and customize this
+"all-in-one" configuration, but this will be covered by a future
+enhancement. There is no intention to allow users to grow these
+single-node installations into multi-node installations.
+
+This enhancement can be seen as a first step of what is described in
+https://github.com/openshift/enhancements/pull/302
+
+## Motivation
+
```

Comment:
This feels like it's missing the use-case?  Is it just for demoing something - I'm not really clear on the why for this...

Reply:
This is a first step for zero touch single node cluster.